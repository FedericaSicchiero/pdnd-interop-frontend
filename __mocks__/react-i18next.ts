import noop from 'lodash/noop'

export const useTranslation = () => {
  return {
    t: (str: string) => str,
    i18n: {
      changeLanguage: () => new Promise(noop),
      language: 'it',
    },
  }
}

export const Trans = ({ children }) => children
