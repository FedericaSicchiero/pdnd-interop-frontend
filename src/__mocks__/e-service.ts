import { EServiceReadType } from '../../types'

export const eserviceDraft: EServiceReadType = {
  producer: {
    id: 'djofsi-sdfjdsi-djsfs',
    name: 'Comune di Bologna',
  },
  name: 'Nome e-service',
  description: 'Descrizione e-service',
  technology: 'REST',
  id: 'sdjof-sdfjdspof-dsfdsjf',
  state: 'DRAFT',
  descriptors: [],
  attributes: {
    verified: [],
    declared: [],
    certified: [
      {
        single: {
          id: 'dsdsld-dsdlds-lsdasdas',
          creationTime: '2022-02-28T16:16:18.879093Z',
          explicitAttributeVerification: false,
          verified: false,
          origin: 'dfkdsfk',
          code: 'dfjdso',
          name: 'Attributo 1',
          description: 'Descrizione attributo 1',
        },
      },
    ],
  },
}
