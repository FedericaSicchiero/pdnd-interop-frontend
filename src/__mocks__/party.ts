import { Party } from '../../types'

export const partyActiveManager: Party = {
  id: 'dsofdm-dsfjds',
  attributes: [],
  state: 'ACTIVE',
  role: 'MANAGER',
  productInfo: { createdAt: '', id: 'interop', role: 'admin' },
  description: 'Comune di Test',
  institutionId: 'dsfjisdo-sdfdjs-sdfjsd',
  digitalAddress: 'comune@test.it',
}

export const partyActiveDelegate: Party = {
  id: 'dsofdm-dsfjds',
  attributes: [],
  state: 'ACTIVE',
  role: 'DELEGATE',
  productInfo: { createdAt: '', id: 'interop', role: 'admin' },
  description: 'Comune di Test',
  institutionId: 'dsfjisdo-sdfdjs-sdfjsd',
  digitalAddress: 'comune@test.it',
}

export const partyActiveOperatorApi: Party = {
  id: 'dsofdm-dsfjds',
  attributes: [],
  state: 'ACTIVE',
  role: 'OPERATOR',
  productInfo: { createdAt: '', id: 'interop', role: 'api' },
  description: 'Comune di Test',
  institutionId: 'dsfjisdo-sdfdjs-sdfjsd',
  digitalAddress: 'comune@test.it',
}

export const partyActiveOperatorSecurity: Party = {
  id: 'dsofdm-dsfjds',
  attributes: [],
  state: 'ACTIVE',
  role: 'OPERATOR',
  productInfo: { createdAt: '', id: 'interop', role: 'security' },
  description: 'Comune di Test',
  institutionId: 'dsfjisdo-sdfdjs-sdfjsd',
  digitalAddress: 'comune@test.it',
}
