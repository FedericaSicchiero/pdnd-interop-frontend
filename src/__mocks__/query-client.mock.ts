import { QueryClient, QueryClientConfig } from '@tanstack/react-query'
import { queryClientConfig } from '../config/query-client'
import { deepmerge } from '@mui/utils'
import noop from 'lodash/noop'

const queryClientConfigMock: QueryClientConfig = deepmerge(
  {
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
    // Disables logging
    logger: {
      log: noop,
      warn: noop,
      error: noop,
    },
  },
  queryClientConfig
)

export const queryClientMock = new QueryClient(queryClientConfigMock)
