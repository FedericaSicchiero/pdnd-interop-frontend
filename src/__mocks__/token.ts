export const token = process.env.REACT_APP_MOCK_TOKEN as string
