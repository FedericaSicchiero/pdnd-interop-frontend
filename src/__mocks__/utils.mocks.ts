import { Paginated } from './../api/comon.api.types'

/** Wraps the data passed in the pagination data structure */
export function createPaginatedMockData<T>(
  data: Array<T>,
  pagination?: Partial<Paginated<T>['pagination']>
): Paginated<T> {
  return {
    results: data,
    pagination: {
      offset: pagination?.offset ?? 0,
      limit: pagination?.offset ?? 50,
      totalResults: pagination?.offset ?? data.length,
    },
  }
}
