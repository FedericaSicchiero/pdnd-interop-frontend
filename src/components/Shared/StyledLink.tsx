import React, { FunctionComponent } from 'react'
import { Link as RouterLink } from 'react-router-dom'
import { Link as MUILink, LinkProps } from '@mui/material'

type StyledLinkProps = {
  to?: string
  component?: any
}

export const StyledLink: FunctionComponent<StyledLinkProps & LinkProps> = React.forwardRef(
  ({ children, to, component = RouterLink, ...props }, ref) => {
    return (
      <MUILink component={component} to={to} {...props} ref={ref}>
        {children}
      </MUILink>
    )
  }
)
