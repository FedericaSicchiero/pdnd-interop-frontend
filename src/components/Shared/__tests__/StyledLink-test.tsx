import React from 'react'
import renderer from 'react-test-renderer'
import { AllTheProviders } from '../../../__mocks__/providers'
import { StyledLink } from '../StyledLink'

describe('Snapshot', () => {
  it('matches link', () => {
    const component = renderer.create(
      <AllTheProviders>
        <StyledLink to="/rotta-esempio">Clicca qui</StyledLink>
      </AllTheProviders>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
