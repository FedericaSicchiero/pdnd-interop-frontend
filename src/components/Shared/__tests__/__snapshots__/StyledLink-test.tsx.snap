// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot matches link 1`] = `
<a
  className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-6xddht-MuiTypography-root-MuiLink-root"
  href="/rotta-esempio"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
>
  Clicca qui
</a>
`;
